### ISSUE TITLE
Error in Pykaldi on Ubuntu

### ISSUE THREAD

**Issue by ppsingh98** [pykaldi/pykaldi#173]
I  tried to install Pykaldi on Ubuntu and encountered the following error
[1/983] Generating iostream-clifwrap.c...-clifwrap.h, iostream-clifwrap-init.cc
FAILED: kaldi/base/iostream-clifwrap.cc kaldi/base/iostream-clifwrap.h kaldi/base/iostream-clifwrap-init.cc 
cd /home/ppsingh/nfr/pykaldi/build/kaldi/base && /home/ppsingh/nfrenv/bin/pyclif --py3output --matcher_bin=/home/ppsingh/nfrenv/clang/bin/clif-matcher --ccdeps_out /home/ppsingh/nfr/pykaldi/build/kaldi/base/iostream-clifwrap.cc --header_out /home/ppsingh/nfr/pykaldi/build/kaldi/base/iostream-clifwrap.h --ccinit_out /home/ppsingh/nfr/pykaldi/build/kaldi/base/iostream-clifwrap-init.cc --modname=kaldi.base._iostream --prepend=clif/python/types.h -I/home/ppsingh/nfr/pykaldi/kaldi/lib -I/home/ppsingh/nfr/pykaldi/kaldi -I/home/ppsingh/nfr/pykaldi/build/kaldi -I/home/ppsingh/nfr/pykaldi/tools/kaldi/src "-f-I/usr/include/python3.6m              -I/home/ppsingh/nfr/pykaldi/kaldi/lib              -I/home/ppsingh/nfr/pykaldi/kaldi              -I/home/ppsingh/nfr/pykaldi/build/kaldi              -I/home/ppsingh/nfr/pykaldi/tools/kaldi/src              -I/home/ppsingh/nfrenv/clang/lib/clang/5.0.0/include               -std=c++11 -I.. -isystem /home/ppsingh/nfr/pykaldi/tools/kaldi/tools/openfst/include -O1 -Wall -Wno-sign-compare -Wno-unused-local-typedefs -Wno-deprecated-declarations -Winit-self -DKALDI_DOUBLEPRECISION=0 -DHAVE_EXECINFO_H=1 -DHAVE_CXXABI_H -DHAVE_ATLAS -I/home/ppsingh/nfr/pykaldi/tools/kaldi/tools/ATLAS_headers/include -msse -msse2 -pthread -g -fPIC -Wno-maybe-uninitialized" /home/ppsingh/nfr/pykaldi/kaldi/base/iostream.clif && /home/ppsingh/nfr/pykaldi/tools/use_namespace.sh /home/ppsingh/nfr/pykaldi/build/kaldi/base/iostream-clifwrap.cc

Line .123456789.123456789.123456789.123456789
   1:"""C++ std::iostream wrapper.\n
   2:\n
   3:This module provides the wrappers needed for working with C++ std::iostream\n
   4:instances. The wrappers for input and/or output stream types do not expose any\n
   5:members other than `good` since we do not want to work with them directly in\n
   6:Python. They can be used for wrapping methods and functions that accept or\n
   7:return iostream objects. The get_* utility functions provide the handles for\n
   8:the standard C++ streams.\n
   9:"""\n
  10:\n
  11:from "base/iostream-ext.h":\n
  12:  namespace `::std`:\n
  13:    class istream:\n
  14:      """Input stream."""\n
  15:      def `good` as good(self) -> bool:\n
  16:        """Checks whether the state of the stream is good."""\n
  17:\n
  18:    class ostream:\n
  19:      """Output stream."""\n
  20:      def `good` as good(self) -> bool:\n
  21:        """Checks whether the state of the stream is good."""\n
  22:\n
  23:    class iostream(istream):\n
  24:      """Input/output stream."""\n
  25:      pass\n
  26:\n
  27:    def `GetStdinPtr` as get_stdin() -> istream:\n
  28:      """Returns standard input stream."""\n
  29:\n
  30:    def `GetStdoutPtr` as get_stdout() -> ostream:\n
  31:      """Returns standard output stream."""\n
  32:\n
  33:    def `GetStderrPtr` as get_stderr() -> ostream:\n
  34:      """Returns standard error stream."""\n
  35:\n
  36:    def `Flush` as flush(os: ostream):\n
  37:      """Flushes output stream."""\n
  38:\n
  39:    def `Read` as read_text(is: istream) -> str:\n
  40:      """Reads and returns the contents of the stream as a unicode string."""\n
  41:\n
  42:    def `Read` as read(is: istream) -> bytes:\n
  43:      """Reads and returns the contents of the stream as a bytes object."""\n
  44:\n
  45:    def `ReadLine` as readline_text(is: istream) -> str:\n
  46:      """Reads and returns a line from the input stream.\n
  47:\n
  48:      If the stream is already at EOF, an empty unicode string is returned.\n
  49:      """\n
  50:\n
  51:    def `ReadLine` as readline(is: istream) -> bytes:\n
  52:      """Reads and returns a line from the input stream.\n
  53:\n
  54:      If the stream is already at EOF, an empty bytes object is returned.\n
  55:      """\n
  56:\n
  57:    def `Write` as write_text(os: ostream, s: str):\n
  58:      """Writes unicode string to output stream."""\n
  59:\n
  60:    def `Write` as write(os: ostream, b: bytes):\n
  61:      """Writes bytes object to output stream."""\n
_ParseError: Expected {Suppress:(lineEnd) | {"from" quoted string, starting with " ending with " Suppress:(":") Suppress:(lineEnd) - indented block} | {Suppress:("from") WordEnd name [. name]... - "import" name} | {Empty Suppress:("from") quoted string, starting with " ending with " Suppress:("import") - Suppress:("*") [{Suppress:("as") - name}]} | {"type" - name Suppress:("=") type} | {"use" - quoted string, starting with ` ending with ` Suppress:("as") - name [. name]...} | {"interface" - name - Suppress:("<") - name [{Suppress:(",") name}]... - Suppress:(">") - Suppress:(":") Suppress:(lineEnd) - indented block}}, found '"'  (at char 0), (line:1, col:1)
[2/983] Generating timer-clifwrap.cc, timer-clifwrap.h, timer-clifwrap-init.cc
FAILED: kaldi/base/timer-clifwrap.cc kaldi/base/timer-clifwrap.h kaldi/base/timer-clifwrap-init.cc 
cd /home/ppsingh/nfr/pykaldi/build/kaldi/base && /home/ppsingh/nfrenv/bin/pyclif --py3output --matcher_bin=/home/ppsingh/nfrenv/clang/bin/clif-matcher --ccdeps_out /home/ppsingh/nfr/pykaldi/build/kaldi/base/timer-clifwrap.cc --header_out /home/ppsingh/nfr/pykaldi/build/kaldi/base/timer-clifwrap.h --ccinit_out /home/ppsingh/nfr/pykaldi/build/kaldi/base/timer-clifwrap-init.cc --modname=kaldi.base._timer --prepend=clif/python/types.h -I/home/ppsingh/nfr/pykaldi/kaldi/lib -I/home/ppsingh/nfr/pykaldi/kaldi -I/home/ppsingh/nfr/pykaldi/build/kaldi -I/home/ppsingh/nfr/pykaldi/tools/kaldi/src "-f-I/usr/include/python3.6m              -I/home/ppsingh/nfr/pykaldi/kaldi/lib              -I/home/ppsingh/nfr/pykaldi/kaldi              -I/home/ppsingh/nfr/pykaldi/build/kaldi              -I/home/ppsingh/nfr/pykaldi/tools/kaldi/src              -I/home/ppsingh/nfrenv/clang/lib/clang/5.0.0/include               -std=c++11 -I.. -isystem /home/ppsingh/nfr/pykaldi/tools/kaldi/tools/openfst/include -O1 -Wall -Wno-sign-compare -Wno-unused-local-typedefs -Wno-deprecated-declarations -Winit-self -DKALDI_DOUBLEPRECISION=0 -DHAVE_EXECINFO_H=1 -DHAVE_CXXABI_H -DHAVE_ATLAS -I/home/ppsingh/nfr/pykaldi/tools/kaldi/tools/ATLAS_headers/include -msse -msse2 -pthread -g -fPIC -Wno-maybe-uninitialized" /home/ppsingh/nfr/pykaldi/kaldi/base/timer.clif && /home/ppsingh/nfr/pykaldi/tools/use_namespace.sh /home/ppsingh/nfr/pykaldi/build/kaldi/base/timer-clifwrap.cc

Line .123456789.123456789.123456789.123456789
   1:\n
   2:from "base/timer.h":\n
   3:  namespace `kaldi`:\n
   4:    class Timer:\n
   5:      """Kaldi timer."""\n
   6:      @add__init__\n
   7:      def `Timer` as new(self, set_timer: bool):\n
   8:        """Creates and optionally resets the timer to current time."""\n
   9:\n
  10:      def `Reset` as reset(self):\n
  11:        """Resets the timer to current time."""\n
  12:\n
  13:      def `Elapsed` as elapsed(self) -> float:\n
  14:        """Returns elapsed time since timer was reset."""\n
_ParseError: invalid statement in indented block, found '"'  (at char 66), (line:5, col:7)

Complete error file is enclosed. Please help to resolve the issue.
[Pykaldi_error.txt](https://github.com/pykaldi/pykaldi/files/3723125/Pykaldi_error.txt)

**Comment by vrmpx** [comment]
Seems like this error is caused by a vanilla clif installation. Please make sure you installed the pykaldi branch of clif (not only the clif in pykaldi).

**Comment by ppsingh98** [comment]
I had installed clif with the script install_clif.sh
This script is doing git clone if clif folder does not exist, else it does git pull. Do i need to clone/download myself.

**Comment by ppsingh98** [comment]
Tried again after cloning clif from pykaldi branch. After this commented the following in install_clif.sh
"git pull"
My environment has Python 3.6.8
Both Clif and Kaldi scripts  generated some warnings and "Done installing clif/kaldi" message at the end.
Execution of  "python setup.py install" again generated the same errors as enclosed in the error file earlier.
Please help to resolve the issue and complete Pykaldi installation.

**Comment by vrmpx** [comment]
I am sorry this is taking so long but it is quite a uncommon problem. 

Just to be absolutely sure, the pyclif you installed from the pykaldi branch was put in `/home/ppsingh/nfrenv/bin/pyclif`, correct?

Could you delete the build directory and try running `DEBUG=1 python setup.py install` again?

**Comment by ppsingh98** [comment]
Thanks for your time. 
Yes, pyclif does exist in /home/ppsingh/nfrenv/bin/pyclif
Deleted the folder pykaldi/build and executed the command:
DEBUG=1 python setup.py install
The error log is enclosed. 
[PyKaldi_debug.txt](https://github.com/pykaldi/pykaldi/files/3743618/PyKaldi_debug.txt)

**Comment by vrmpx** [comment]
Hey @ppsingh98, 
I didn't find any problems with the installation in Ubuntu 18.04. I got it installed inside a docker container, and my only issue was some confusion caused by multiple python versions installed in the system. Nothing a simple update-alternatives couldn't handle though. Albeit this was a clean system where I only ran the installation scripts we provide, getting LLVM from SVN instead of git.

So I am sorry you couldn't get it to run on your system. Have you thought of using our [docker image](https://github.com/pykaldi/pykaldi/tree/master/docker) or [anaconda packages](https://anaconda.org/Pykaldi/pykaldi/) as alternatives?
